flang/lib/Optimizer/Runtime/Assign.cpp

```diff
@@ -0,0 +1,26 @@
+//===-- Assign.cpp -- generate assignment runtime API calls ---------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "flang/Optimizer/Runtime/Assign.h"
+#include "../../../runtime/assign.h"
+#include "flang/Optimizer/Builder/FIRBuilder.h"
+#include "flang/Optimizer/Runtime/RTBuilder.h"
+
+using namespace Fortran::runtime;
+
+void fir::runtime::genAssign(fir::FirOpBuilder &builder, mlir::Location loc,
+                             mlir::Value destBox, mlir::Value sourceBox) {
```

Comment:
Do `destBox` and `sourceBox` have to be of `!fir.box<>` type? Should be assert that here?

Reply:
Yes, I added a comment in the header file regarding what these arguments must be. I think an assert would be redundant on top of mlir verification that will complain if the something that is not a fir.box<> is converted to it while generating the genAssign runtime call (I actually tend to prefer having the mlir errors in such cases and be able to dump the MLIR module and directly see what is happening).